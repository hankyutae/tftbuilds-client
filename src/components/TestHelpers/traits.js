let traits =
  { "Void": { "desc": "Basic Attacks and Spells from select Void champions deal true damage.<br><br><row>(@MinUnits@) 1 Random Void champion</row><br><row>(@MinUnits@) All Void champions</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Void.png", "name": "Void", "effects": [{ "vars": [{ "name": "PercentArmorPen", "value": 100 }, { "name": "CB42693F", "value": 1 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "PercentArmorPen", "value": 100 }, { "name": "CB42693F", "value": 1 }], "maxUnits": 25000, "minUnits": 4 }], "isOrigin": true }, "Wild": { "desc": "Select allies' Basic Attacks are empowered to grant +@AttackSpeedPercent@% Bonus Attack Speed (stacks up to 5 times).<br><br><row>(@MinUnits@) Wild Allies Only</row><br><row>(@MinUnits@) All Allies, and allies' basic attacks cannot be dodged.</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Wild.png", "name": "Wild", "effects": [{ "vars": [{ "name": "AttackSpeedPercent", "value": 12 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "AttackSpeedPercent", "value": 12 }], "maxUnits": 25000, "minUnits": 4 }], "isOrigin": true }, "Demon": { "desc": "Basic Attacks from Demons have a @ManaDrainProcChance@% chance to reduce their target's Mana by @ManaDrainBurn@ and restore Mana to the Demon.<br><br><expandRow>(@MinUnits@) @ManaDrainDrain@ Mana restored</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Demon.png", "name": "Demon", "effects": [{ "vars": [{ "name": "ManaDrainProcChance", "value": 40 }, { "name": "ManaDrainBurn", "value": 20 }, { "name": "ManaDrainDrain", "value": 15 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "ManaDrainProcChance", "value": 40 }, { "name": "ManaDrainBurn", "value": 20 }, { "name": "ManaDrainDrain", "value": 30 }], "maxUnits": 5, "minUnits": 4 }, { "vars": [{ "name": "ManaDrainProcChance", "value": 40 }, { "name": "ManaDrainBurn", "value": 20 }, { "name": "ManaDrainDrain", "value": 45 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": true }, "Exile": { "desc": "<row>(@MinUnits@) If an Exile has no adjacent allies at the start of combat, they gain a shield equal to @ShieldPercent@% of their maximum Health</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Exile.png", "name": "Exile", "effects": [{ "vars": [{ "name": "ShieldPercent", "value": 100 }], "maxUnits": 25000, "minUnits": 1 }], "isOrigin": true }, "Ninja": { "desc": "Ninjas gain bonus Attack Damage and Spell Power.<br><br>This trait is only active when you have exactly 1 or 4 unique Ninjas.<br><br><row>(@MinUnits@) Ninja gains +@ADValue@ Attack Damage & +@APValue@% Spell Power</row><br><row>(@MinUnits@) All Ninjas gain +@ADValue@ Attack Damage & +@APValue@% Spell Power</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Ninja.png", "name": "Ninja", "effects": [{ "vars": [{ "name": "ADValue", "value": 50 }, { "name": "APValue", "value": 50 }], "maxUnits": 1, "minUnits": 1 }, { "vars": [{ "name": "ADValue", "value": 80 }, { "name": "APValue", "value": 80 }], "maxUnits": 4, "minUnits": 4 }], "isOrigin": true }, "Noble": { "desc": "At the start of combat, select allies gain +@Armor@ Armor, +@MagicResist@ Magic Resistance, and heal @Heal@ Health on-hit.<br><br><row>(@MinUnits@) 1 Random Ally</row><br><row>(@MinUnits@) All Allies</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Noble.png", "name": "Noble", "effects": [{ "vars": [{ "name": "Armor", "value": 50 }, { "name": "Heal", "value": 30 }, { "name": "MagicResist", "value": 50 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "Armor", "value": 50 }, { "name": "Heal", "value": 30 }, { "name": "MagicResist", "value": 50 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": true }, "Robot": { "desc": "(@MinUnits@) Robots start combat at full Mana", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Robot.png", "name": "Robot", "effects": [{ "vars": [], "maxUnits": 25000, "minUnits": 1 }], "isOrigin": true }, "Dragon": { "desc": "<row>(@MinUnits@) Dragons gain @MagicReduction@% resistance to Magic Damage</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Dragon.png", "name": "Dragon", "effects": [{ "vars": [{ "name": "MagicReduction", "value": 75 }], "maxUnits": 25000, "minUnits": 2 }], "isOrigin": true }, "Knight": { "desc": "All allies block a flat amount of damage from all sources.<br><br><expandRow>(@MinUnits@) @BlockAmount@ Damage Blocked</expandRow>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Knight.png", "name": "Knight", "effects": [{ "vars": [{ "name": "BlockAmount", "value": 15 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "BlockAmount", "value": 35 }], "maxUnits": 5, "minUnits": 4 }, { "vars": [{ "name": "BlockAmount", "value": 65 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": false }, "Pirate": { "desc": "<row>(@MinUnits@) At the end of each combat against another player, a treasure chest spawns on your arena containing up to 4 gold</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Pirate.png", "name": "Pirate", "effects": [{ "vars": [], "maxUnits": 25000, "minUnits": 3 }], "isOrigin": true }, "Ranger": { "desc": "Every @ProcRate@ seconds, Rangers have a chance to double their Attack Speed for @ProcRate@ seconds. <br><br><expandRow>(@MinUnits@) @ProcPercent@% Chance to Double Attack Speed</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Ranger.png", "name": "Ranger", "effects": [{ "vars": [{ "name": "ProcPercent", "value": 25 }, { "name": "ASMultiplier", "value": 2 }, { "name": "ProcRate", "value": 3 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "ProcPercent", "value": 70 }, { "name": "ASMultiplier", "value": 2 }, { "name": "ProcRate", "value": 3 }], "maxUnits": 25000, "minUnits": 4 }], "isOrigin": false }, "Yordle": { "desc": "Yordles gain a chance to dodge enemy Basic Attacks.<br><br><expandRow>(@MinUnits@) @MissPercent@% Chance to Dodge</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Yordle.png", "name": "Yordle", "effects": [{ "vars": [{ "name": "MissPercent", "value": 30 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "MissPercent", "value": 60 }], "maxUnits": 8, "minUnits": 6 }, { "vars": [{ "name": "MissPercent", "value": 90 }], "maxUnits": 25000, "minUnits": 9 }], "isOrigin": true }, "Brawler": { "desc": "Brawlers gain bonus Maximum Health.<br><br><expandRow>(@MinUnits@) @BonusHealth@ Bonus Health</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Brawler.png", "name": "Brawler", "effects": [{ "vars": [{ "name": "BonusHealth", "value": 250 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "BonusHealth", "value": 500 }], "maxUnits": 5, "minUnits": 4 }, { "vars": [{ "name": "BonusHealth", "value": 900 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": false }, "Glacial": { "desc": "Basic Attacks from Glacials have a chance to stun their target for @StunDuration@ seconds.<br><br><expandRow>(@MinUnits@) @ProcPercent@% Chance to Stun</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Glacial.png", "name": "Glacial", "effects": [{ "vars": [{ "name": "ProcPercent", "value": 20 }, { "name": "StunDuration", "value": 1.5 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "ProcPercent", "value": 33 }, { "name": "StunDuration", "value": 1.5 }], "maxUnits": 5, "minUnits": 4 }, { "vars": [{ "name": "ProcPercent", "value": 50 }, { "name": "StunDuration", "value": 1.5 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": true }, "Hextech": { "desc": "At the start of combat, an ally Hextech launches and detonates a pulse bomb, temporarily disabling nearby enemy items for @DisableDuration@ seconds.<br><br><row>(@MinUnits@) @HexSize@ Hex Radius</row><br><row>(@MinUnits@) @HexSize@ Hex Radius</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Hextech.png", "name": "Hextech", "effects": [{ "vars": [{ "name": "DisableDuration", "value": 7 }, { "name": "HexSize", "value": 1 }, { "name": "A004F927", "value": 1.25 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "DisableDuration", "value": 7 }, { "name": "HexSize", "value": 2 }, { "name": "A004F927", "value": 1.25 }], "maxUnits": 25000, "minUnits": 4 }], "isOrigin": true }, "Phantom": { "desc": "<row>(@MinUnits@) At the start of combat, a random enemy's Health is set to @HPToSet@</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Phantom.png", "name": "Phantom", "effects": [{ "vars": [{ "name": "HPToSet", "value": 100 }], "maxUnits": 25000, "minUnits": 2 }], "isOrigin": true }, "Assassin": { "desc": "Innate: At the start of combat, Assassins leap to the farthest enemy.<br><br>Assassins gain bonus Critical Strike Damage and Chance.<br><br><expandRow>(@MinUnits@) +@CritAmpPercent@% Critical Strike Damage & +@CritChanceAmpPercent@% Critical Strike Chance</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Assassin.png", "name": "Assassin", "effects": [{ "vars": [{ "name": "CritAmpPercent", "value": 75 }, { "name": "CritChanceAmpPercent", "value": 10 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "CritAmpPercent", "value": 150 }, { "name": "CritChanceAmpPercent", "value": 20 }], "maxUnits": 8, "minUnits": 6 }, { "vars": [{ "name": "CritAmpPercent", "value": 225 }, { "name": "CritChanceAmpPercent", "value": 30 }], "maxUnits": 25000, "minUnits": 9 }], "isOrigin": false }, "Guardian": { "desc": "<row>(@MinUnits@) At the start of combat, Guardians grant +@Armor@ Armor to adjacent allies (this bonus can stack any number of times)</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Guardian.png", "name": "Guardian", "effects": [{ "vars": [{ "name": "Armor", "value": 35 }], "maxUnits": 25000, "minUnits": 2 }], "isOrigin": false }, "Imperial": { "desc": "At the start of combat, select Imperials are empowered to deal double damage.<br><br><row>(@MinUnits@) 1 Random Imperial</row><br><row>(@MinUnits@) All Imperials</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Imperial.png", "name": "Imperial", "effects": [{ "vars": [{ "name": "DamageMultiplier", "value": 2 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "DamageMultiplier", "value": 2 }], "maxUnits": 25000, "minUnits": 4 }], "isOrigin": true }, "Sorcerer": { "desc": "Innate: Sorcerers gain double Mana from Basic Attacks.<br><br>All allies have increased Spell Power.<br><br><expandRow>(@MinUnits@) +@AP@% Spell Power</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Sorcerer.png", "name": "Sorcerer", "effects": [{ "vars": [{ "name": "AP", "value": 40 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "AP", "value": 100 }], "maxUnits": 8, "minUnits": 6 }, { "vars": [{ "name": "AP", "value": 175 }], "maxUnits": 25000, "minUnits": 9 }], "isOrigin": false }, "Gunslinger": { "desc": "Gunslinger Basic Attacks have a @ProcPercent@%  chance to fire additional bullets at other enemies within range.<br><br>These bullets deal damage like Basic Attacks and apply on-hit effects.<br><br><row>(@MinUnits@) Attack @ExtraTargetsHit@ extra enemies</row><br><row>(@MinUnits@) Attack @ExtraTargetsHit@ extra enemies</row><br><row>(@MinUnits@) Attack @ExtraTargetsHit@ extra enemies</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Gunslinger.png", "name": "Gunslinger", "effects": [{ "vars": [{ "name": "ProcPercent", "value": 50 }, { "name": "F768E185", "value": 1 }, { "name": "ExtraTargetsHit", "value": 1 }], "maxUnits": 3, "minUnits": 2 }, { "vars": [{ "name": "ProcPercent", "value": 50 }, { "name": "F768E185", "value": 2 }, { "name": "ExtraTargetsHit", "value": 2 }], "maxUnits": 5, "minUnits": 4 }, { "vars": [{ "name": "ProcPercent", "value": 50 }, { "name": "F768E185", "value": 3 }, { "name": "ExtraTargetsHit", "value": 3 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": false }, "Blademaster": { "desc": "Blademaster Basic Attacks have a @ProcChance@% chance to trigger additional attacks against their target.<br><br>These additional attacks deal damage like Basic Attacks and trigger on-hit effects.<br><br><row>(@MinUnits@) One Additional Attack</row><br><row>(@MinUnits@) Two Additional Attacks</row><br><row>(@MinUnits@) Four Additional Attacks</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Blademaster.png", "name": "Blademaster", "effects": [{ "vars": [{ "name": "ProcChance", "value": 45 }, { "name": "ExtraAttacks", "value": 1 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "ProcChance", "value": 45 }, { "name": "ExtraAttacks", "value": 2 }], "maxUnits": 8, "minUnits": 6 }, { "vars": [{ "name": "ProcChance", "value": 45 }, { "name": "ExtraAttacks", "value": 4 }], "maxUnits": 25000, "minUnits": 9 }], "isOrigin": false }, "Elementalist": { "desc": "Innate: Elementalists gain double mana from Basic Attacks.<br><br><row>(@MinUnits@) At the start of combat, summon a Golem with 2200 Health to fight alongside your allies</row>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Elementalist.png", "name": "Elementalist", "effects": [{ "vars": [{ "name": "HPMultiplier" }, { "name": "558229DF" }, { "name": "SizeCoefficient" }], "maxUnits": 25000, "minUnits": 3 }], "isOrigin": false }, "Shapeshifter": { "desc": "Shapeshifters gain bonus Maximum Health when they transform.<br><br><expandRow>(@MinUnits@) @HPPercent@% Bonus Maximum Health</expandRow><br>", "icon": "ASSETS/UX/TraitIcons/Trait_Icon_Shapeshifter.png", "name": "Shapeshifter", "effects": [{ "vars": [{ "name": "HPPercent", "value": 60 }], "maxUnits": 5, "minUnits": 3 }, { "vars": [{ "name": "HPPercent", "value": 100 }], "maxUnits": 25000, "minUnits": 6 }], "isOrigin": false } };

export default traits;