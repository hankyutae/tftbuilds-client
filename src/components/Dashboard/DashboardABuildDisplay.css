.build-display-in-dashboard{
  position:relative;
  display:flex;
  flex-wrap:wrap;
  align-items:top;
  background-color:whitesmoke;
}
.build-display-in-dashboard-box{
  position:relative;
  background-color:whitesmoke;
  margin:10px 0px;
}
.load-and-share{
  display:flex;
  flex-direction: column;
  justify-content: start;
  align-items:left;
}
.dash-delete-build{
  position:absolute;
  top:0;
  right:0;
}
.build-share-link{ 
  overflow-wrap: break-word;
  word-wrap: break-word;

  -ms-word-break: break-all;
  /* This is the dangerous one in WebKit, as it breaks things wherever */
  word-break: break-all;
  /* Instead use this non-standard one: */
  word-break: break-word;
}