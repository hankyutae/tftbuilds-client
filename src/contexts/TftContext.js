import React from 'react';

export const TftContext = React.createContext({
  error: null,
  arrayOfPrices:[1,2,3,4,5],
  arrayOfStars:['bronze','silver','gold'],
  expandedTraits:{},
  numOfBasicItems:9,
  loginState:0,
  toggleLoginState :  ()=>{ },
  setError: () => { },
  clearError: () => { },
  setChampions: () => { },
  setItems: () => { },
  setTraits: () => { },
  newCurrentBuild: () => { },
  addChampToCurrentBuild: () => { },
  removeChampFromCurrentBuild: ()=>{},
  changeStarsOnChamp:() =>{},
  lowerStarsOnChamp:() =>{},
  raiseStarsOnChamp: () =>{},
  addItemToChamp:() =>{},
  removeItemFromChamp:() =>{},
  switchItemOnChamp:() =>{},
  toggleExpandedOnCurrentBuild:()=>{},
  toggleStatsOnCurrentBuild:()=>{},
  toggleAddItemOnCurrentBuild:()=>{},
  toggleUltOnCurrentBuild:()=>{},
  toggleTraitDesc:()=>{},
  champions: {},
  items: {},
  traits: {},
  currentBuild: []
});

export default TftContext;

export class TftProvider extends React.Component {
  state = {
    arrayOfPrices:[1,2,3,4,5,7],
    arrayOfStars:['bronze','silver','gold'],
    numOfBasicItems:9,
    champions: this.props.champions||{},
    items: this.props.items||{},
    traits: this.props.traits||{},
    currentBuild:this.props.currentBuild||[],
    expandedTraits:this.props.expandedTraits ||{},
    loginState:0,
    toggleLoginState:()=>{
      console.log('even made it here');
      this.setstate({
        loginState:(this.state.loginState+1)
      })
    },
    toggleTraitDesc:(trait)=>{
      let newTraitList={...this.state.expandedTraits};
      newTraitList[trait]=!newTraitList[trait];
      this.setState({
        expandedTraits:newTraitList
      })
    },
    toggleExpandedOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isExpanded:!(newBuild[index].isExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleStatsOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isStatsExpanded:!(newBuild[index].isStatsExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleAddItemOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isAddItemExpanded:!(newBuild[index].isAddItemExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleUltOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isUltExpanded:!(newBuild[index].isUltExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },

    changeStarsOnChamp:(index, stars)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:stars};
      this.setState({
        currentBuild: newBuild
      })
    },
    lowerStarsOnChamp:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:newBuild[index].stars<=1? 1: newBuild[index].stars-1};
      this.setState({
        currentBuild: newBuild
      })
    },
    raiseStarsOnChamp:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:newBuild[index].stars>=this.state.arrayOfStars.length? this.state.arrayOfStars.length: newBuild[index].stars+1};
      this.setState({
        currentBuild: newBuild
      })
    },

    addItemToChamp:(index,item) =>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.push(item);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    removeItemFromChamp:(index,indexOfItem) =>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.splice(indexOfItem,1);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    switchItemOnChamp:(index,item,itemIndex)=>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.splice(itemIndex,1,item);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    removeChampFromCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild.splice(index,1);
      this.setState({
        currentBuild: newBuild
      })
    },

    addChampToCurrentBuild:(champ)=>{
      let curtChamp={
        id:champ.id,
        stars:champ.stars? champ.stars: 1,
        items:champ.items? champ.items: [],
      };/* 
      curtChamp.stars=1;
      curtChamp.items=[]; */
      this.setState({
        currentBuild: [...this.state.currentBuild,curtChamp]
      })
    },

    setCurrentBuild: (newBuild) => {
      this.setState({ currentBuild: newBuild })
    },

    setChampions: champions => {
      this.setState({ champions })
    },

    setItems: items => {
      this.setState({ items })
    },

    setTraits: traits => {
      this.setState({ traits })
    },

    setError: error => {
      console.error(error)
      this.setState({ error })
    },

    clearError: () => {
      this.setState({ error: null })
    },

  };

  render() {
    /* const value = {
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setTraits: this.setTraits,
      setItems: this.setItems,
      setChampions: this.setChampions,
      items: this.items,
      traits: this.traits,
      champions: this.champions
    } */
    return (
      <TftContext.Provider value={this.state}>
        {this.props.children}
      </TftContext.Provider>
    )
  }
}