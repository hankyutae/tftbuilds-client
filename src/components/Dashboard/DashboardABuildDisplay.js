import React from 'react';
import './DashboardABuildDisplay.css'
import ImgLink from '../../services/create-img-link-service';
import { Link } from 'react-router-dom';
import TftContext from '../../contexts/TftContext';
import TftBuildsApiService from '../../services/tftbuilds-api-service';

export default class DashboardABuildDisplay extends React.Component {
  state = {
    isExpanded: false,
    isRevealed: false,
    copySuccess: '',
    isDeleting:false,
  }
  static contextType = TftContext
  handleLoadBuildClick = (e) => {
    e.stopPropagation();
    this.context.setCurrentBuild(this.props.build['build_data']);
  }
  componentDidUpdate(prevProps){
    /* this.setState({
      isExpanded: false,
      isRevealed: false,
      copySuccess: '',
      isDeleting:false,
    }) */
  }
  handleBuildClick = (e) => {
    e.stopPropagation();
    this.setState({
      isExpanded: !this.state.isExpanded
    })
  }
  handleDeleteClick = (e) => {
    e.stopPropagation();
    TftBuildsApiService.deleteBuild(this.props.build.id)
      .then(res => {
        this.setState({
          isExpanded: false,
          isRevealed: false,
          copySuccess: '',
          isDeleting:false,
        },()=>{
          this.props.handleRemove(this.props.index);
        })
      })
  }
  toggleDeleting = (e)=>{
    this.setState({
      isDeleting:!this.state.isDeleting
    })
  }
  handleMakePublicClick = (e) => {
    e.stopPropagation();
    if (this.props.build['is_public']) {
      this.setState({
        isRevealed: true
      })
    }
    else {
      TftBuildsApiService.makeBuildPublic(this.props.build.id)
        .then(res => {
          this.setState({
            isRevealed: true
          },this.props.handleMakePublic(this.props.index))
        })
    }
  }
  copyLink =(e)=>{
    this.textArea.select();
    document.execCommand('copy');
    this.setState({copySuccess:'Copied!'})
  }
  render() {
    return (
      <div className='build-display-in-dashboard-box'>
        {Object.keys(this.context.champions).length !== 0 && Object.keys(this.context.items).length !== 0 &&
          <>
            <div className='dashboard-build-top'>
              <div className='load-and-share'>
                <div className='load-build-button button-hover-lighten'>
                  <Link className='nav-link' to='/create-build' onClick={this.handleLoadBuildClick}> Load Build</Link>
                </div>
                <div className='share-build-button button-hover-lighten' onClick={this.handleMakePublicClick}>
                  Share
                </div>
              </div>
              <div className='dash-delete-build'>
                {
                  this.state.isDeleting? 
                  <div className='delete-yes-or-no-container'>
                    <div className='yes-delete-build-button deleting-yes-or-no-button button-hover-lighten'onClick={this.handleDeleteClick}>
                      Delete
                    </div>
                    <div className='no-delete-build-button deleting-yes-or-no-button button-hover-lighten' onClick={this.toggleDeleting}>
                      No
                    </div>
                  </div>
                  :
                  <div className='red delete-build-button button-hover-lighten' onClick={this.toggleDeleting}>Delete </div>
                }
                
              </div>

            </div>
            <div className='dashboard-build-bottom'>

            
              <div className='build-display-in-dashboard' onClick={this.handleBuildClick}>
                {




                  this.props.build['build_data'].filter((champ, index) => index < 11).map((champ, index) => {
                    return (
                      <div key={index} className='dash-build-champ-details'>
                        <img key={index} className='dash-build-champ-icon' src={ImgLink.createLink(this.context.champions[champ.id].splash)} alt={this.context.champions[champ.id].name + ' icon'} />
                        <div className='dash-box-champ-items'>
                          {champ.items.map((item,index)=>{
                            return <img key={index} className='dash-build-item-icon' src={ImgLink.createLink(this.context.items[item].icon)} alt={this.context.items[item].name + ' icon'} />
                          })}
                        </div>
                      </div>
                    )
                  })

                }
              </div>

              {this.state.isRevealed &&
                <div className='build-link-container'>
                  <textarea 
                    className='build-share-link' 
                    ref={(textarea) => this.textArea = textarea} 
                    value={`${window.location.origin}/build/${this.props.build.id}`}
                    readOnly
                  />
                  {
                    document.queryCommandSupported('copy') &&
                    <div className='copy-link-button' onClick={this.copyLink}>
                      Copy
                    </div>
                  }
                  {
                    this.state.copySuccess &&
                    <span className='copy-success'>
                      {this.state.copySuccess}
                    </span>
                  }

                </div>
              }
            </div>
          </>
        }
      </div>
    )
  }
}