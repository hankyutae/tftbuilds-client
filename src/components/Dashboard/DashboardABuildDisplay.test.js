import React from 'react';
import ReactDOM from 'react-dom';
import DashboardABuildDisplay from './DashboardABuildDisplay';
import {BrowserRouter} from 'react-router-dom';
import {TftProvider} from '../../contexts/TftContext';
import champions from '../TestHelpers/champions';
import items from '../TestHelpers/items';
import traits from '../TestHelpers/traits';
import renderer from "react-test-renderer";

let build={
  build_data: [{"id": 38, "items": [16, 15, 5], "stars": 1}, {"id": 103, "items": [5, 17, 25], "stars": 3}, {"id": 25, "items": [], "stars": 1}, {"id": 136, "items": [], "stars": 1}, {"id": 30, "items": [], "stars": 1}, {"id": 421, "items": [], "stars": 1}, {"id": 31, "items": [], "stars": 1}],
  date_created:'',
  date_modified:'',
  is_public:false,
  user_id:''
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <TftProvider champions={champions} items={items} traits={traits} >
      <BrowserRouter>
        <DashboardABuildDisplay build={build} index={0} />
      </BrowserRouter>
    </TftProvider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <TftProvider champions={champions} items={items} traits={traits} >
        <BrowserRouter>
          <DashboardABuildDisplay build={build} index={0} />
        </BrowserRouter>
      </TftProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
