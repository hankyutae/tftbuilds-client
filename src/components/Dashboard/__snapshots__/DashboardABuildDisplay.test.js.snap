// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="build-display-in-dashboard-box"
>
  <div
    className="dashboard-build-top"
  >
    <div
      className="load-and-share"
    >
      <div
        className="load-build-button button-hover-lighten"
      >
        <a
          className="nav-link"
          href="/create-build"
          onClick={[Function]}
        >
           Load Build
        </a>
      </div>
      <div
        className="share-build-button button-hover-lighten"
        onClick={[Function]}
      >
        Share
      </div>
    </div>
    <div
      className="dash-delete-build"
    >
      <div
        className="red delete-build-button button-hover-lighten"
        onClick={[Function]}
      >
        Delete 
      </div>
    </div>
  </div>
  <div
    className="dashboard-build-bottom"
  >
    <div
      className="build-display-in-dashboard"
      onClick={[Function]}
    >
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Kassadin icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_kassadin.png"
        />
        <div
          className="dash-box-champ-items"
        >
          <img
            alt="Bloodthirster icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_bloodthirster.tft.png"
          />
          <img
            alt="Guardian Angel icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_guardianangel.tft.png"
          />
          <img
            alt="Chain Vest icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_chainvest.tft.png"
          />
        </div>
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Ahri icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_ahri.png"
        />
        <div
          className="dash-box-champ-items"
        >
          <img
            alt="Chain Vest icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_chainvest.tft.png"
          />
          <img
            alt="Zeke's Herald icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_zekesherald.tft.png"
          />
          <img
            alt="Phantom Dancer icon"
            className="dash-build-item-icon"
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_phantomdancer.tft.png"
          />
        </div>
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Morgana icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_morgana.png"
        />
        <div
          className="dash-box-champ-items"
        />
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Aurelion Sol icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_aurelionsol.png"
        />
        <div
          className="dash-box-champ-items"
        />
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Karthus icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_karthus.png"
        />
        <div
          className="dash-box-champ-items"
        />
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Rek'Sai icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_reksai.png"
        />
        <div
          className="dash-box-champ-items"
        />
      </div>
      <div
        className="dash-build-champ-details"
      >
        <img
          alt="Cho'Gath icon"
          className="dash-build-champ-icon"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_chogath.png"
        />
        <div
          className="dash-box-champ-items"
        />
      </div>
    </div>
  </div>
</div>
`;
